### DISCUSSION TITLE
Where to subscribe to actions and events in the pipeline?

### DISCUSSION THREAD

**Discussion by NGumby** [thomasgalliker/Plugin.FirebasePushNotifications#88] [General]
On Android, I got everything working when the app is in foreground or background.  But when the app is killed, actions are not working, and NotificationDeleted is also not working.  I get these logs :
HandleNotificationAction drops event "NotificationAction" (no event subscribers / no queue present).
HandleNotificationDeleted drops event "NotificationDeleted" (no event subscribers / no queue present).

Should this code be moved in UseFirebasePushNotifications?

Or is it necessary to use a PersistentQueueFactory?

**Comment by thomasgalliker** [comment]
Good point. I think it's a long time since I tested with a killed Android app.
